ehsm_kms_service/function.js

```diff
@@ -0,0 +1,273 @@
+const crypto = require('crypto')
+
+const logger = require('./logger')
+const { MAX_TIME_STAMP_DIFF, TIMESTAMP_LEN } = require('./constant')
+const { ehsm_kms_params } = require('./ehsm_kms_params.js')
+const apis = require('./apis')
+
+const _result = (code, msg, data = {}) => {
+  return {
+    code: code,
+    message: msg,
+    result: {
+      ...data,
+    },
+  }
+}
+// base64 encode
+const base64_encode = (str) => new Buffer.from(str).toString('base64')
+
+// base64 decode
+const base64_decode = (base64_str) =>
+  new Buffer.from(base64_str, 'base64').toString()
+
+/**
+ * The parameters of non empty parameter values in set sign_parmas are sorted from small
+ * to large according to the ASCII code of the parameter name (dictionary order),
+ * and the format of URL key value pairs (i.e. key1 = value1 & key2 = Value2...)
+ * is spliced into a string
+ * @param {object} sign_parmas
+ * @returns string
+ */
+const params_sort_str = (sign_parmas) => {

```

Comment:
sign_parmas   typo

Reply:
Sorry, I don't understand the purpose of this function: params_sort_str.  we can discuss.

Reply:
it will be fix

One of the sign generation rules:
 - Parameter names are sorted by ASCII code from small to large (dictionary order)
params_sort_str do this sort rule

Reply:
it will be fix

Reply:
One of the sign generation rules:
 - Parameter names are sorted by ASCII code from small to large (dictionary order)
params_sort_str do this sort rule